### ISSUE TITLE
GetHashEntryLocale comments

### ISSUE THREAD

**Issue by WareditorGit** [ladislav-zezula/StormLib#259]
In GetHashEntryLocale():
```c++
// Storm_2016.dll: 150209D9
// If (locale matches or is neutral) OR (platform matches or is neutral), remember this as the best entry
// Also remember the first matching entry for Starcraft maps
if(pHash->lcLocale == 0 || pHash->lcLocale == lcLocale)
{
    if(pHash->Platform == 0 || pHash->Platform == Platform)
    {
        pBestEntry = pHash;
    }
}
```

The comment says "OR" but the the best entry only get updated when (locale matches or is neutral) AND (platform matches or is neutral). I am not sure if the comment is wrong or the code is wrong but it has been like that since this code was added in 2016.
The comment also states `remember the first matching entry for Starcraft maps` but it doesn't do anything in the code about that.

I am also curious about another comment just above:
```c++
// Only do that for non-0 locale&platform, because for loc&plat=0, there's different
// processing in Warcraft III vs. Starcraft, which is abused by some protectors.
```
In a previous commit it used to also say: `some Warcraft III maps have several items with neutral locale&platform, which leads to wrong item being returned`. Is my understanding correct that, in Warcraft III,  for neutral locale&platform the correct entry to return is the last matching one?

**Comment by ladislav-zezula** [comment]
>The comment says "OR" but the the best entry only get updated when (locale matches or is neutral)
> AND (platform matches or is neutral).

The comment only applies to the next line. 

> Is my understanding correct that, in Warcraft III, for neutral locale&platform the correct entry to return is the last matching one?

Yes. Warcraft III returns the last matching entry.

**Comment by ladislav-zezula** [comment]
Please, just let it go. I spent quite lot of time with tuning this algorithm to match Warcraft III and all the god-damn MPQ twisters, so caller protectors. If the comment is wrong, then you can live with it. Or, prepare a pull request for updated comment.

**Comment by WareditorGit** [comment]
The question was to understand if the comment is wrong or the code which seemed like something important but you disagree.

**Comment by ladislav-zezula** [comment]
I fixed the comment: 7a664bf6be4377aabe03f118fcdb6d1d3abc48f5
